datastore.go

```diff
@@ -96,18 +104,26 @@ func (a *accessor) Get(key ds.Key) (value []byte, err error) {
 }
 
 func (a *accessor) Has(key ds.Key) (exists bool, err error) {
+	a.closeLk.RLock()
+	defer a.closeLk.RUnlock()
 	return a.ldb.Has(key.Bytes(), nil)
 }
 
-func (d *accessor) GetSize(key ds.Key) (size int, err error) {
-	return ds.GetBackedSize(d, key)
+func (a *accessor) GetSize(key ds.Key) (size int, err error) {
+	a.closeLk.RLock()
+	defer a.closeLk.RUnlock()
```

Comment:
Shouldn't be necessary. `GetBackedSize` can only access public functions on this datastore.